### PR TITLE
fix(maven): Use consistent precision for extended ranges

### LINKED ISSUE #6620: Maven: preserve range precision
Hi @rarkins,

**What Renovate type are you using?**

+ Platform : https://github.com
+ Renovate : https://renovatebot.com

**Describe the bug**

As I am using `[1.8,1.9)` in https://github.com/the-benchmarker/web-frameworks/blob/839da5f0c8b7baf2aad1e36873bc0cfb470a74c6/java/act/pom.xml#L16, meaning from `1.8` to `1.9` where **1.9** is excluded, I expected that `renovatebot` `PR` keeps the logic.

I mean the above range precise the **minor** interval. As such, I expect `renovatebot` to create `PR` with :
+ `[1.9,1.10)`
+ `[1.10,1.11)`
+ `[1.11,1.12)`
+ ...

Regards,

### PR DESCRIPTION
**Pull request by zharinov**
Closes #6620

### CODE REVIEW

**Comment by zharinov** [comment]
Example: https://github.com/zharinov/renovate-test-maven/pull/2

**Review by viceice — DISMISSED** [review]

File: lib/versioning/maven/index.spec.ts [hunk 1]
```diff
@@ -251,15 +251,30 @@ describe('versioning/maven/compare', () => {
...
+      ['[0.21,0.22)', '0.23', '[0.22,0.23)'],
+
+      ['[1.8,1.9)', '1.9.0.1', '[1.8,1.9)'],
+      ['[1.8a,1.9)', '1.9.0.1', '[1.8a,1.9)'],
+      ['[1.8,1.9.0)', '1.9.0.1', '[1.8,1.9.0)'],
+      ['[1.8,1.9.0.0)', '1.9.0.1', '[1.8,1.9.0.1)'],
+      ['[1.8,1.9.0.0)', '1.10.1', '[1.8,1.10.1.0)'],
+
+      ['[1.8,1.9)', '1.9.1', '[1.8,1.9)'],
+      ['[1.8,1.9)', '1.10.0', '[1.9,1.10)'],
```
[the remark this answers is not in the document]
Comment (zharinov) [reply lib/versioning/maven/index.spec.ts:271]: Looks better in this context. What upgrade for `[1.8, 1.9]` and `1.10.0` would be better:`[1.8, 1.10]` or `[1.9, 1.10]`? Btw probably we need to think about range strategies other than `auto` for maven.

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/versioning/maven/index.spec.ts:271]: I think it should be 1.11)

[the remark this answers is not in the document]
Comment (zharinov) [reply lib/versioning/maven/index.spec.ts]: Okay, take a look again, I've made a slightly different implementation

**Pull request merged**